commands/azureCommands/create-registry.ts

```diff
@@ -0,0 +1,210 @@
+    } while (!location);
+    return location;
+}
+
+async function acquireResourceGroup(subscription: SubscriptionModels.Subscription): Promise<ResourceGroup> {
+    //Acquire each subscription's data simultaneously
+    let resourceGroup;
+    let resourceGroupName;
+    const resourceGroupClient = new ResourceManagementClient(AzureCredentialsManager.getInstance().getCredentialByTenantId(subscription.tenantId), subscription.subscriptionId);
+    let resourceGroups = await AzureCredentialsManager.getInstance().getResourceGroups(subscription);
+
+    let resourceGroupNames: string[] = [];
+    resourceGroupNames.push('+ Create new resource group');
+    for (let resGroupName of resourceGroups) {
+        resourceGroupNames.push(resGroupName.name);
+    }
+
+    do {
+        resourceGroupName = await vscode.window.showQuickPick(resourceGroupNames, { 'canPickMany': false, 'placeHolder': 'Choose a Resource Group to be used' });
+        if (resourceGroupName === undefined) { throw new Error('user Exit'); }
+        if (resourceGroupName === '+ Create new resource group') {
+            let loc = await acquireLocation(resourceGroup, subscription);
+            resourceGroupName = await createNewResourceGroup(loc, resourceGroupClient);
+        }
+        resourceGroups = await AzureCredentialsManager.getInstance().getResourceGroups(subscription);
+        resourceGroup = resourceGroups.find(resGroup => { return resGroup.name === resourceGroupName; });
+
+        if (!resourceGroupName) { vscode.window.showErrorMessage('You must select a valid resource group'); }
+    } while (!resourceGroupName);
+    return resourceGroup;
+}
+
+async function createNewResourceGroup(loc: string, resourceGroupClient: ResourceManagementClient): Promise<string> {
+    let promptMessage = 'Resource group name?';
+
+    let opt: vscode.InputBoxOptions = {
+        ignoreFocusOut: false,
+        prompt: promptMessage
+    };
+
+    let resourceGroupName: string;
+    let resourceGroupStatus: boolean;
+
+    while (opt.prompt) {
+        resourceGroupName = await vscode.window.showInputBox(opt);
+        resourceGroupStatus = await resourceGroupClient.resourceGroups.checkExistence(resourceGroupName);
+        if (!resourceGroupStatus) {
+            opt.prompt = null;
+            console.log("true status, prompt message = null");
+        } else {
+            opt.prompt = "That resource group name is already in existence. Try again: ";
+            console.log("false status, prompt message = try again");
+        }
+    }
+
+    let newResourceGroup: ResourceGroup = {
+        name: resourceGroupName,
+        location: loc,
+    };
+
+    //Potential error when two clients try to create same resource group name at once
+    try {
+        await resourceGroupClient.resourceGroups.createOrUpdate(resourceGroupName, newResourceGroup);
+    } catch (error) {
+        vscode.window.showErrorMessage("That resource group name is already in existence. Try again");
```

Comment:
>That resource group name is already in existence. Try again [](start = 40, length = 59)

Help the user identify the input error 
$"The resource group  '{newResourceGroup}' already exists"